Enhance abstract skill for DITA short descriptions (#4814)
theforeman/foreman-documentation: .cursor/skills/abstract/SKILL.md

---
name: abstract

> description: Review or write an abstract (DITA short description) for a documentation module

Reviewer: ```suggestion
description: Review or write an abstract (also called a short description) for a documentation module
```
This docs set is written in AsciiDoc and there are no plans to introduce DITA.

Abstracts, also called short descriptions, help readers and AI-powered search tools find the information that they need and confirm that they are in the right place.

## Instructions

When reviewing or writing the abstract, follow these principles:

- Do not simply repeat the heading of the module; build upon it.
- Avoid self-referential language (for example: avoid "This procedure..." or "This module...").
- Avoid feature-centric language (for example: avoid "This feature...").
- Use user-centric language; address the user directly as "you".
- Follow these length constraints: 50-300 characters, 1-2 sentences, a single paragraph.
- Do not use sentence fragments. Always use full sentences.
- Write one sentence per line.
- For module-type-specific abstract rules, look up the reference file that matches the `:_mod-docs-content-type:` AsciiDoc attribute in the module (open and apply that file in full):
  - `CONCEPT` → [references/concept.md](references/concept.md)
  - `PROCEDURE` → [references/procedure.md](references/procedure.md)
  - `REFERENCE` → [references/reference.md](references/reference.md)

> ## Examples of good abstracts

Author: I should probably move examples to the references.

Abstracts, also called short descriptions, help readers and AI-powered search tools find the information that they need and confirm that they are in the right place.

> Abstracts typically contain the following information:

Reviewer: Can you add this back? These lines still apply as a generic description/definition of what an abstract does. Is there a need to remove them?

Author: I've removed this generic instruction in favor of the specific ones. But sure, I will put it back.

When reviewing or writing the abstract, follow these principles:

- Do not simply repeat the heading of the module; build upon it.
- Avoid self-referential language (for example: avoid "This procedure...", "This module...", "This table...").
- Avoid feature-centric language (for example: avoid "This feature...").
- Do not use sentence fragments.
- When needed, address the user as "you".
- Follow these length constraints: 50-300 characters, 1-2 sentences, a single paragraph.

> - Write one sentence per line.

Reviewer: I'd like to see these style skills focused strictly on what they're doing: in this case, recording requirements for abstracts.

```suggestion
```

For the one-sentence-per-line rule, perhaps a separate skill based on our contribution guidelines would be a better fit (should we convert our whole CONTRIBUTING.md into a skill? 🤔) but since that's probably not an option right now, personally I just have this particular instruction saved in my own local settings :)

Reviewer 2: @aneta-petrova Honestly is there any meaningful difference between a set of guidelines like our Contributing and a skill? You can reference either from Claude at least...

Reviewer: I've been reading up on how to best organize internal project documentation and instructions. There are skills, rules, CLAUDE.md, and other features, each designed with a specific purpose in mind. A useful article, for example, is https://code.claude.com/docs/en/memory.

In case of this particular line, though, I don't think it's a good idea to commit a general rule (which we should apply everywhere) to a specific skill. Because then each skill would need to include this rule. And potentially other general rules.

Based on what I know, this particular rule would be a better fit for a CLAUDE.md file. Or CONTRIBUTING.md.

Making sure we record instructions in the right files would also help us build something useful on top of these skills and AI-friendly contributing instructions, like I'm trying to do in https://github.com/theforeman/foreman-documentation/pull/4830

Author: Okay, I agree, this would be a good candidate for a [rule](https://cursor.com/docs/rules).

When reviewing or writing the abstract, follow these principles:

- Do not simply repeat the heading of the module; build upon it.
- Avoid self-referential language (for example: avoid "This procedure...", "This module...", "This table...").
- Avoid feature-centric language (for example: avoid "This feature...").
- Do not use sentence fragments.

> - When needed, address the user as "you".

Reviewer: What exactly does "When needed" refer to? Perhaps you meant:
```suggestion
- When addressing the user, address them as "you".
```

When reviewing or writing the abstract, follow these principles:

- Ensure the abstract explains the What and the Why as defined in the section "Definition of an abstract".

>   - For the `What` part, state the action or definition clearly.

Author: ```suggestion
  - For the `What` part, state the action or concept clearly.
```
